code/sarix_model/retrospective-experiments/sarix_experiments.py

```diff
@@ -0,0 +1,34 @@
+# This script should be run with code/sarix_model as the working directory:
+# python retrospective-experiments/sarix_experiments.py
+
+import os
+import datetime
+from multiprocessing import Pool
+from itertools import product
+
+# model_names = [
+#     f'sarix_p{p}_4rt_theta{theta_pooling}_sigmanone_xmas_spike' \
+#         for (p, theta_pooling) in product([2, 4, 6, 8], ['none', 'shared'])
+# ]
+model_names = [
+    f'sarix_p{p}_4rt_theta{theta_pooling}_sigmanone' \
+        for (p, theta_pooling) in product([2, 4, 6, 8], ['none', 'shared'])
+]
+
+ref_dates = [
+    (datetime.date(2023, 10, 14) + datetime.timedelta(i * 7)).isoformat() \
+        for i in range(30)
+]
+
+def run_command(command):
+    """Run system command"""
+    os.system(command)
+
+
+output_root = '../../retrospective-hub/model-output'
+
+commands = [f'python sarix_model.py --ref_date {ref_date} --output_root {output_root} --model_name {model_name}' \
+                for (model_name, ref_date) in product(model_names, ref_dates)]
```

Comment:
`subprocess` is the recommended python module for running system commands (it's more secure, is more consistent across platforms, and is more flexible to work with in terms of handling command output and errors).

`subprocess.run` accepts a command list where each component of the command is an individual item. So if you refactor your list of commands along the lines of the suggestion below, you could swap in subprocess without too much effort (and could explore [its parameters](https://docs.python.org/3/library/subprocess.html#subprocess.run) should you ever need to reroute the output or add error handling)

```suggestion
def run_command(command):
    """Run system command"""
    subprocess.run(command)


output_root = '../../retrospective-hub/model-output'

commands = [[
    "python sarix_model.py",
    "--ref-date", ref_date,
    "--output_root", output_root,
    "--model_name", model_name]
    for (model_name, ref_date) in product(model_names, ref_dates)]
```